modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/SlimFrameworkServerCodegen.java

```diff
@@ -0,0 +1,195 @@
+        typeMapping.put("string", "string");
+        typeMapping.put("byte", "int");
+        typeMapping.put("boolean", "boolean");
+        typeMapping.put("date", "DateTime");
+        typeMapping.put("datetime", "DateTime");
+        typeMapping.put("file", "string");
+        typeMapping.put("map", "map");
+        typeMapping.put("array", "array");
+        typeMapping.put("list", "array");
+        typeMapping.put("object", "object");
+
+        supportingFiles.add(new SupportingFile("README.mustache", packagePath.replace('/', File.separatorChar), "README.md"));
+        supportingFiles.add(new SupportingFile("composer.json", packagePath.replace('/', File.separatorChar), "composer.json"));
+        supportingFiles.add(new SupportingFile("index.mustache", packagePath.replace('/', File.separatorChar), "index.php"));
+        supportingFiles.add(new SupportingFile(".htaccess", packagePath.replace('/', File.separatorChar), ".htaccess"));
+    }
+
+    public CodegenType getTag() {
+        return CodegenType.SERVER;
+    }
+
+    public String getName() {
+        return "slim";
+    }
+
+    public String getHelp() {
+        return "Generates a Slim Framework server library.";
+    }
+
+    @Override
+    public String escapeReservedWord(String name) {
+        return "_" + name;
+    }
+
+    @Override
+    public String apiFileFolder() {
+        return (outputFolder + "/" + apiPackage()).replace('/', File.separatorChar);
+    }
+
+    public String modelFileFolder() {
+        return (outputFolder + "/" + modelPackage()).replace('/', File.separatorChar);
+    }
+
+    @Override
+    public String getTypeDeclaration(Property p) {
+        if (p instanceof ArrayProperty) {
+            ArrayProperty ap = (ArrayProperty) p;
+            Property inner = ap.getItems();
+            return getSwaggerType(p) + "[" + getTypeDeclaration(inner) + "]";
+        } else if (p instanceof MapProperty) {
+            MapProperty mp = (MapProperty) p;
+            Property inner = mp.getAdditionalProperties();
+            return getSwaggerType(p) + "[string," + getTypeDeclaration(inner) + "]";
+        }
+        return super.getTypeDeclaration(p);
+    }
+
+    @Override
+    public String getSwaggerType(Property p) {
+        String swaggerType = super.getSwaggerType(p);
+        String type = null;
+        if (typeMapping.containsKey(swaggerType)) {
+            type = typeMapping.get(swaggerType);
+            if (languageSpecificPrimitives.contains(type)) {
+                return type;
+            } else if (instantiationTypes.containsKey(type)) {
+                return type;
+            }
+        } else {
+            type = swaggerType;
+        }
+        if (type == null) {
+            return null;
+        }
+        return toModelName(type);
+    }
+
+    public String toDefaultValue(Property p) {
+        return "null";
+    }
+
+
+    @Override
+    public String toVarName(String name) {
```

Comment:
For variable and method parameter name, what about using camelCase instead? Ref: https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#6-closures uses camelCase for variable naming.